packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -20,24 +24,80 @@ use tokio::time::{sleep, timeout};
 
 use crate::{exsrv::Handler, process};
 
+/// Configuration values used for the [`hash_engine`] subprocess used.
+///
+/// See the [`process`] module for more information.
+///
+/// [`Process`]: crate::process::Process
 #[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "clap", derive(clap::Args))]
 pub struct ExperimentConfig {
-    pub num_workers: usize,
+    /// Project output path folder.
+    ///
+    /// The folder will be created if it's missing.
+    #[cfg_attr(
+        feature = "clap",
+        clap(short, long, default_value = "./output", env = "HASH_OUTPUT")
+    )]
+    pub output: PathBuf,
+
+    /// Output format emitted to the terminal.
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "pretty", arg_enum, env = "HASH_EMIT")
+    )]
     pub emit: OutputFormat,
-    pub output_folder: PathBuf,
+
+    /// Output location where to emit logs.
+    ///
+    /// Can be `stdout`, `stderr` or any file name. Relative to `--log-folder` if a file is
+    /// specified.
+    #[cfg_attr(feature = "clap", clap(long, default_value = "stderr"))]
     pub output_location: OutputLocation,
+
+    /// Logging output folder.
+    #[cfg_attr(feature = "clap", clap(long, default_value = "./log"))]
     pub log_folder: PathBuf,
-    pub engine_start_timeout: Duration,
-    pub engine_wait_timeout: Duration,
+
+    /// Engine start timeout in seconds.
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "2", env = "ENGINE_START_TIMEOUT")
+    )]
+    pub start_timeout: u64,
+
+    /// Engine wait timeout in seconds.
+    #[cfg_attr(
+        feature = "clap",
+        clap(long, default_value = "60", env = "ENGINE_WAIT_TIMEOUT")
+    )]
+    pub wait_timeout: u64,
+
+    /// Max number of parallel workers (must be power of 2).
```

Comment:
The comment about the worker count needs to be a power of two comes from before I changed the CLI to use clap. My guess is, that nothing happens, but we are also setting the number of workers to `1` currently as it's unimplemented. @litvand Do we need this restriction? If so, I'll add a check, if not, I'll remove the comment.

> Do we expect users to know what "workers" are at this stage?

No, I don't think, this is required. This would need elaboration on the runners as well and I think it's out of scope for a CLI documentation.

> Do we want to say that by default it's set to num_cpus?

Yes!